clumio/resource_organizational_unit.go

```diff
@@ -0,0 +1,307 @@
+	if description != "" {
+		request.Description = &description
+	}
+	users := getStringSlice(d, "users")
+	request.Users = users
+	//TODO: Handle entities after the entityModel is made public in the SDK
+	res, apiErr := orgUnitsAPI.CreateOrganizationalUnit(nil, request)
+	if apiErr != nil {
+		return diag.Errorf(
+			"Error creating Clumio OrganizationalUnit. Error: %v", string(apiErr.Response))
+	}
+	d.SetId(*res.Id)
+	return clumioOrganizationalUnitRead(ctx, d, meta)
+}
+
+// clumioOrganizationalUnitRead handles the Read action for the Clumio OrganizationalUnit Resource.
+func clumioOrganizationalUnitRead(
+	_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*apiClient)
+	orgUnitsAPI := orgUnits.NewOrganizationalUnitsV1(client.clumioConfig)
+	res, apiErr := orgUnitsAPI.ReadOrganizationalUnit(d.Id())
+	if apiErr != nil {
+		if strings.Contains(apiErr.Error(), "The resource is not found.") {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf(
+			"Error creating Clumio AWS Connection. Error: %v", string(apiErr.Response))
+
+	}
+	err := d.Set("children_count", int(*res.ChildrenCount))
+	if err != nil {
+		return diag.Errorf(
+			"Error setting children_count schema attribute. Error: %v", err)
+	}
+	err = d.Set("user_count", int(*res.UserCount))
+	if err != nil {
+		return diag.Errorf("Error setting user_count schema attribute. Error: %v", err)
+	}
+	if res.ConfiguredDatasourceTypes != nil {
+		configuredDatasourceTypes := make([]string, 0)
+		for _, dsType := range res.ConfiguredDatasourceTypes {
+			configuredDatasourceTypes = append(configuredDatasourceTypes, *dsType)
+		}
+		err = d.Set("configured_datasource_types", configuredDatasourceTypes)
+		if err != nil {
+			return diag.Errorf(
+				"Error setting configured_datasource_types schema attribute."+
+					" Error: %v", err)
+		}
+	}
+	if res.DescendantIds != nil {
+		descendantIds := make([]string, 0)
+		for _, dsType := range res.ConfiguredDatasourceTypes {
+			descendantIds = append(descendantIds, *dsType)
+		}
+		err = d.Set("descendant_ids", descendantIds)
+		if err != nil {
+			return diag.Errorf(
+				"Error setting descendant_ids schema attribute. Error: %v", err)
+		}
+	}
+	return nil
+}
+
+// clumioOrganizationalUnitUpdate handles the Update action for the Clumio OrganizationalUnit Resource.
+func clumioOrganizationalUnitUpdate(
+	ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if d.HasChange("email") {
+		return diag.Errorf("email is not allowed to be changed")
+	}
+	client := meta.(*apiClient)
+	orgUnitsAPI := orgUnits.NewOrganizationalUnitsV1(client.clumioConfig)
+	updateRequest := &models.PatchOrganizationalUnitV1Request{}
+	if d.HasChange("description") {
+		description := getStringValue(d, "description")
+		updateRequest.Description = &description
+	}
+	if d.HasChange("name") {
+		name := getStringValue(d, "name")
+		updateRequest.Name = &name
+	}
```

Comment:
Curious why do we need to check for `HasChange` here? I assume given the API looks like a PATCH, things like "description" and "name" are optional values and so if given they should go through?